sympy/vector/implicitregion.py

```diff
@@ -0,0 +1,308 @@
+        # Now it iterates over a range and tries to find a point on the region.
+        equation = self.equation
+
+        if len(self.variables) == 1:
+            return (list(solveset(equation, self.variables[0], domain=S.Reals))[0],)
+        elif len(self.variables) == 2:
+            x, y = self.variables
+
+            for x_reg in range(-100, 100):
+                if len(solveset(equation.subs(x, x_reg), self.variables[1], domain=S.Reals)) > 0:
+                    return (x_reg, list(solveset(equation.subs(x, x_reg), domain=S.Reals))[0])
+        elif len(self.variables) == 3:
+            x, y, z = self.variables
+
+            for x_reg in range(-10, 10):
+                for y_reg in range(-10, 10):
+                    if len(solveset(equation.subs({x: x_reg, y: y_reg}), self.variables[2], domain=S.Reals)) > 0:
+                        return (x_reg, y_reg, list(solveset(equation.subs({x: x_reg, y: y_reg})))[0])
+
+        if len(self.singular_points()) != 0:
+            return list[self.singular_points()][0]
+
+        raise NotImplementedError()
+
+    def singular_points(self):
+        """
+        Returns a set of singular points of the region.
+
+        The singular points are those points on the region
+        where all partial derivatives vanish.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import x, y
+        >>> from sympy.vector import ImplicitRegion
+        >>> I = ImplicitRegion((x, y), (y-1)**2 -x**3 + 2*x**2 -x)
+        >>> I.singular_points()
+        FiniteSet((1, 1))
+
+        """
+        eq_list = [self.equation]
+        for var in self.variables:
+            eq_list += [diff(self.equation, var)]
+
+        return nonlinsolve(eq_list, list(self.variables))
+
+    def multiplicity(self, point):
+        """
+        Returns the multiplicity of a singular point on the region.
+
+        A singular point (x,y) of region is said to be of multiplicity m
+        if all the partial derivatives off to order m - 1 vanish there.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import x, y, z
+        >>> from sympy.vector import ImplicitRegion
+        >>> I = ImplicitRegion((x, y, z), x**2 + y**3 - z**4)
+        >>> I.singular_points()
+        FiniteSet((0, 0, 0))
+        >>> I.multiplicity((0, 0, 0))
+        2
+
+        """
+        if isinstance(point, Point):
+            point = point.args
+
+        modified_eq = self.equation
+
+        for i, var in enumerate(self.variables):
+            modified_eq = modified_eq.subs(var, var + point[i])
+        modified_eq = expand(modified_eq)
+
+        if len(modified_eq.args) != 0:
+            terms = modified_eq.args
+            m = min([total_degree(term) for term in terms])
+        else:
+            terms = modified_eq
+            m = total_degree(terms)
+
+        return m
+
+    def rational_parametrization(self, parameter1='t', parameter2='s', reg_point=None):
```

Comment:
Is this supposed to work for 2D only? What about `parameters = ('t', 's')` instead?

Reply:
It will work for curves and surfaces given that they are monoid.

Reply:
The report which I followed had algorithms only for curves and surfaces. No Volume regions or higher dimensional region.

>>> What about parameters = ('t', 's') instead?
This looks better. I will fix it.

Reply:
> Is this supposed to work for 2D only? What about parameters = ('t', 's') instead?

It does not work for 3-D curves. It works for 3-D surfaces but does not support higher-dimensional surfaces..